pisa/core/binning.py

```diff
@@ -2730,6 +2730,7 @@ def __getitem__(self, index):
             for d in self.iterdims():
                 if d.name == index:
                     return d
+            raise ValueError(f"index '{index}' not in {[d.name for d in self.iterdims()]}")
```

Comment:
You can use the `names` attr here rather than having to iterate/list-comprehend yourself:

```python
raise ValueError(f"index '{index}' not in {self.names}")
```

... but this is certainly a useful exception to have for users, so thanks for adding it.